changelog/23079.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+secrets/azure: Added note that environment variables take precedence over API config.
```

Comment:
I don't think we need a CL entry for docs changes. Typically only for product changes.

Reply:
+1